### DISCUSSION TITLE
Spread out tighly clustered nodes

### DISCUSSION THREAD

**Discussion by bostrt** [graphia-app/graphia#29] [Q&A] [answered]
I have group of nodes that are not easy to see and seem to glitch about while layout is running. Is there any way to address this in Graphia?

![Screenshot from 2021-09-14 22-33-51](https://user-images.githubusercontent.com/215290/133371355-19405f0f-ca6b-4478-94e5-3b1a4211f008.png)

BTW, love the app!

**Comment by timangus** [comment, marked as answer]
There's a few ways you can tackle this. Firstly simply pause the layout (I assume you've already found this). Unfortunately force directed algorithms sometimes get into oscillating states and it's tricky to detect when that happens. Secondly, you could open the layout settings and try adjusting the forces to spread the nodes out a bit and/or reduce volatility. Finally, you might try an edge reduction transform such as k-NN. This is a process that selectively removes edges in such a way that graph structure is (to some extent) maintained. In doing so, hairballs such as this can typically be expanded. Of course, depending on your application, removing edges may not be acceptable, but I thought I'd mention it anyway. The other potential solution when you have densely packed nodes is to employ an edge contraction transform to combine nodes together based on an edge condition. Again, application dependent.

HTH.
